tests/internal/e2e/test/specs/vcast_coded_test_completion.test.ts

```diff
@@ -0,0 +1,416 @@
+    const tab = (await editorView.openEditor(
+      "manager-template.cpp"
+    )) as TextEditor;
+    await browser.keys([Key.Ctrl, Key.Space]);
+    const contentAssist = await tab.toggleContentAssist(true);
+    await tab.setTextAtLine(14, "// vmock");
+
+    // Ensure cursor is positioned after "// vmock"
+    let currentLine = await tab.getLineOfText("// vmock");
+    await tab.typeTextAt(currentLine, "// vmock".length + 1, " ");
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    // Validate content assist items
+    console.log("Validating content assist for '// vmock'");
+    expect(await contentAssist.hasItem("database")).toBe(true);
+    expect(await contentAssist.hasItem("manager")).toBe(true);
+    expect(await contentAssist.hasItem("Prototype-Stubs")).toBe(true);
+
+    await selectItem(contentAssist, "manager");
+    await tab.typeTextAt(currentLine, "// vmock manager".length + 1, " ");
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    expect(await contentAssist.hasItem("Manager::AddIncludedDessert")).toBe(
+      true
+    );
+    expect(await contentAssist.hasItem("Manager::AddPartyToWaitingList")).toBe(
+      true
+    );
+    expect(await contentAssist.hasItem("Manager::ClearTable")).toBe(true);
+    expect(await contentAssist.hasItem("Manager::GetCheckTotal")).toBe(true);
+    expect(await contentAssist.hasItem("Manager::GetNextPartyToBeSeated")).toBe(
+      true
+    );
+    expect(await contentAssist.hasItem("Manager::PlaceOrder")).toBe(true);
+
+    await selectItem(contentAssist, "Manager::ClearTable");
+
+    await tab.typeTextAt(
+      currentLine,
+      "// vmock manager Manager::ClearTable".length + 1,
+      " "
+    );
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    // Use normalized expected content for validation
+    const contentAssistItems = await contentAssist.getItems();
+    let itemFound = false;
+
+    // If the completion item is too long it is encoded weridly vrom vscode with "⏎". --> Normalize it.
+    for (const item of contentAssistItems) {
+      const label = await item.getLabel();
+      if (
+        normalizeContentAssistString(label) ===
+        normalizeContentAssistString(normalizedExpectedFunctionOutput)
+      ) {
+        itemFound = true;
+        await selectItem(contentAssist, normalizedExpectedFunctionOutput);
+        break;
+      }
+    }
+
+    expect(itemFound).toBe(true);
+    console.log("Content assist validation passed.");
+  });
+});
+
+/**
+ * Function to select an item from the content assistant.
+ * @param contentAssist Content assist including the autocompletion items.
+ * @param item String consisting of the item label.
+ */
+async function selectItem(contentAssist: ContentAssist, item: string) {
```

Comment:
`selectItem` and `normalizeContentAssist` should be moved to the existing `utils.ts`, as they might be useful for some other tests.

Reply:
Moved `selectItem `and `normalizeContentAssist ` to `vcast_utils.ts` in b797c7864c92738222537ffd5e50d5fc216835ba